sqlmesh/core/loader.py

```diff
@@ -452,17 +464,29 @@ def _load_materializations(self) -> None:
                 if os.path.getsize(path):
                     import_python_file(path, context_path)
 
-    def _load_signals(self) -> None:
+    def _load_signals(self) -> UniqueKeyDict[str, signal]:
         """Loads signals for the built-in scheduler."""
+
+        signals_max_mtime: t.Optional[float] = None
+
         for context_path, config in self._context.configs.items():
             for path in self._glob_paths(
                 context_path / c.SIGNALS,
                 ignore_patterns=config.ignore_patterns,
                 extension=".py",
             ):
                 if os.path.getsize(path):
+                    self._track_file(path)
+                    signal_file_mtime = self._path_mtimes[path]
+                    signals_max_mtime = (
```

Comment:
Are we actually using this value anywhere? Shouldn't we use this to invalidate the cache?